src/json.rs

```diff
@@ -50,6 +50,136 @@ pub enum JsonValue {
     Null,
 }
 
+impl JsonValue {
+    /// Returns a boolean indicating whether this value is an object or not.
+    pub fn is_object(&self) -> bool {
+        match self {
+            JsonValue::Object(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns a reference to the key-value vec if this value is an object, otherwise returns None.
+    pub fn as_object(&self) -> Option<&[(Vec<char>, JsonValue)]> {
+        match self {
+            JsonValue::Object(obj) => Some(obj),
+            _ => None,
+        }
+    }
+
+    /// Returns the wrapped object if the value is an object, otherwise panics.
+    pub fn to_object(self) -> JsonObject {
```

Comment:
how useful are those to_xxx methods? as_xxx().unwrap() should achieve the same goal right?
I will prefer to avoid shipping footgun in the code.

Reply:
Actually no, because the as_xxx methods return an `Option`, whereas the to_xxx methods return the wrapped value directly.

I could modify this to make it also return an `Option`, but I think it would make it a bit less ergonomic, since there are situations where people know with 100% certainty that they can't get anything other than the wrapped value back.

Reply:
Another idea I had was to introduce the into_xxx methods, which would use the current behaviour of the to_xxx methods. The to_xxx methods would then return an `Option` instead.

Reply:
Right. In that case I think we could just make to_xxx return Option / Result?